code/modules/clothing/suits/armor.dm

```diff
@@ -381,8 +395,50 @@
 			if(M == owner)
 				continue
 			owner.Beam(M,icon_state="lightning[rand(1, 12)]",icon='icons/effects/effects.dmi',time=5)
-			M.adjustFireLoss(25)
+			M.adjustFireLoss(20)
 			playsound(M, 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		disable(rand(2, 5)) // let's not have buckshot set it off 4 times and do 80 burn damage.
+		return 1
+
+/obj/item/clothing/suit/armor/reactive/repulse
+	name = "reactive repulse armor"
+	desc = "An experimental suit of armor that violently throws back attackers with the power of a gravitational anomaly core."
+	///How strong the reactive armor is for throwing
+	var/repulse_power = 3
+	/// How far away are we finding things to throw
+	var/repulse_range = 5
+
+/obj/item/clothing/suit/armor/reactive/repulse/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return 0
+	if(prob(hit_reaction_chance))
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], converting the attack into a wave of force!</span>")
+		var/list/thrownatoms = list()
+		var/atom/throwtarget
+		var/distfromuser
+		for(var/turf/T in range(repulse_range, owner)) //Done this way so things don't get thrown all around hilariously.
+			for(var/atom/movable/AM in T)
+				thrownatoms += AM
+
+		for(var/am in thrownatoms)
+			var/atom/movable/AM = am
+			if(AM == owner || AM.anchored)
+				continue
+
+			throwtarget = get_edge_target_turf(owner, get_dir(owner, get_step_away(AM, owner)))
+			distfromuser= get_dist(owner, AM)
+			if(distfromuser == 0)
+				if(isliving(AM))
+					var/mob/living/M = AM
+					M.Weaken(3)
+					to_chat(M, "<span class='userdanger'>You're slammed into the floor by [owner]'s reactive armor!</span>")
+			else
+				if(isliving(AM))
+					var/mob/living/M = AM
+					to_chat(M, "<span class='userdanger'>You're thrown back by the [owner]'s reactive armor!</span>")
+				spawn(0)
+					AM.throw_at(throwtarget, ((clamp((repulse_power - (clamp(distfromuser - 2, 0, distfromuser))), 3, repulse_power))), 1)//So stuff gets tossed around at the same time.
```

Comment:
Probably want `INVOKE_ASYNC()` here instead, since `spawn()` can cause some problems.
```suggestion
				INVOKE_ASYNC(AM, .proc/throw_at, ((clamp((repulse_power - (clamp(distfromuser - 2, 0, distfromuser))), 3, repulse_power))), 1)//So stuff gets tossed around at the same time.
```
_(doing this on my phone so double check that this actually works)_